From 89fc7765fe2ce15f7fd3aa779dd152fee6eae117 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@gmail.com>
Date: Sat, 22 Aug 2026 07:10:53 +0200
Subject: [PATCH 30/33] mmc: sunxi: Name cleared IDMA status bits

Replace the raw status value with the existing bit definitions.

Signed-off-by: Jernej Skrabec <jernej.skrabec@gmail.com>
---
 drivers/mmc/host/sunxi-mmc.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sunxi-mmc.c b/drivers/mmc/host/sunxi-mmc.c
index a850e85b386e..cd47dd51ccfc 100644
--- a/drivers/mmc/host/sunxi-mmc.c
+++ b/drivers/mmc/host/sunxi-mmc.c
@@ -189,6 +189,11 @@
 #define SDXC_IDMAC_NORMAL_INTERRUPT_SUM		BIT(8)
 #define SDXC_IDMAC_ABNORMAL_INTERRUPT_SUM	BIT(9)
 #define SDXC_IDMAC_HOST_ABORT_INTERRUPT		BIT(10)
+#define SDXC_IDMAC_STATUS_CLEAR \
+	(SDXC_IDMAC_TRANSMIT_INTERRUPT | SDXC_IDMAC_RECEIVE_INTERRUPT | \
+	 SDXC_IDMAC_FATAL_BUS_ERROR | SDXC_IDMAC_DESTINATION_INVALID | \
+	 SDXC_IDMAC_CARD_ERROR_SUM | SDXC_IDMAC_NORMAL_INTERRUPT_SUM | \
+	 SDXC_IDMAC_ABNORMAL_INTERRUPT_SUM)
 #define SDXC_IDMAC_IDLE				(0 << 13)
 #define SDXC_IDMAC_SUSPEND			(1 << 13)
 #define SDXC_IDMAC_DESC_READ			(2 << 13)
@@ -575,7 +580,7 @@ static irqreturn_t sunxi_mmc_finalize_request(struct sunxi_mmc_host *host)
 	}
 
 	if (data) {
-		mmc_writel(host, REG_IDST, 0x337);
+		mmc_writel(host, REG_IDST, SDXC_IDMAC_STATUS_CLEAR);
 		mmc_writel(host, REG_DMAC, 0);
 		rval = mmc_readl(host, REG_GCTRL);
 		rval |= SDXC_DMA_RESET;
-- 
2.43.0

